app/models/manageiq/providers/openstack/cloud_manager/vm.rb

```diff
@@ -19,6 +19,51 @@ def provider_object(connection = nil)
     connection.servers.get(ems_ref)
   end
 
+  def associate_floating_ip(public_network, port = nil)
+    ext_management_system.with_provider_connection(:service => "Network",
+                                                   :tenant_name => cloud_tenant.name) do |connection|
+      unless port
+        network_ports.each do |network_port|
+          # Cycle through all ports and find one that is actually connected to the public network with router,
+          if network_port.public_networks.detect { |x| x.try(:ems_ref) == public_network.ems_ref }
+            port = network_port
+            break
+          end
+        end
+      end
+      raise MiqException::MiqNetworkPortNotDefinedError, "Neutron port for floating IP association is not defined." unless port
+
+      connection.create_floating_ip(public_network.ems_ref, :port_id => port.ems_ref)
+    end
+  end
+
+  def delete_floating_ips(floating_ips)
+    # TODO(lsmola) we have the method here because we need to take actual cloud_tenant from the VM.
+    # This should be refactored to FloatingIP, when we can take tenant from elsewhere, Like user
+    # session? They have it in session in Horizon, ehich correlates the teannt in keytsone token.
+    ext_management_system.with_provider_connection(:service => "Network",
+                                                   :tenant_name => cloud_tenant.name) do |connection|
+
+      floating_ips.each do |floating_ip|
+        begin
+          connection.delete_floating_ip(floating_ip.ems_ref)
+        rescue StandardError => e
+          # The FloatingIp could have been deleted by another process
+          _log.info("Could not delete floating IP #{floating_ip}. Error: #{e}")
```

Comment:
You should include some VM info and/or EMS info in this log message.  Consider the situation where I have more than one OpenStack Provider in an appliance.  Pointing me to the correct OpenStack will help a lot in the logs.